### PR TITLE
Fix / 6907 WP Dashboard Submenu Not Visible

### PR DESCRIPTION
**Pull request by 10upsimon**
## Summary

Addresses issue:

- #6907

## Relevant technical choices

- Added custom z-indexes section to SASS variables storage file at `assets/sass/config/_variables.scss`.
- Defined new `$z-index-sk-header` SASS variable to reference value `9980`.
- Updated the SK Header CSS selector for `.googlesitekit-header` in `assets/sass/components/global/_googlesitekit-header.scss` and changed the existing `z-index` property value to be the newly defined `$z-index-sk-header` variable, thus reducing the z-index to `9980` and no longer causing display issues with the WordPress admin sidebar sub/flyout menus.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/sass/config/_variables.scss [hunk 1]
```diff
@@ -311,3 +311,6 @@ $mdc-text-field-label: $c-interactive-on-disable-container;
 
 // Custom max width for settings notice content
 $max-width-settings-notice-content: 1098px;
+
+// Custom z-index values
```
Comment (techanvil) [inline assets/sass/config/_variables.scss:315]: As @aaemnnosttv alluded to [here](https://github.com/google/site-kit-wp/issues/6907#issuecomment-1634979154), it's probably worth adding a bit more info as a comment to describe the notable, related z-indexes that you found while investigating this issue. No need to go into too much detail - we can always refer back to this issue for the full writeup - but a brief summary would be handy, essentially these bits:

> * The top admin bar and its child menus, at z-index `99999`
> * The left most main/side admin menu wrapper, at z-index `9990`
> * The left most main/side admin menu sub menus on hover, at z-index `9999`

**Review by techanvil — CHANGES_REQUESTED** [review]
Hey @10upsimon, this looks great, I've left a request for a bit more info in the comments. Please take a look.

**Comment by 10upsimon** [comment]
> Hey @10upsimon, this looks great, I've left a request for a bit more info in the comments. Please take a look.

Hey @techanvil , thanks for the feedback. Please see https://github.com/google/site-kit-wp/pull/7332/commits/421532b879570a1161be7ef0504b43ce90e0ff30 where I have added more supporting info via a SASS comment block, I trust this is what you meant.

Thanks!

**Comment by techanvil** [comment]
> > Hey @10upsimon, this looks great, I've left a request for a bit more info in the comments. Please take a look.
> 
> Hey @techanvil , thanks for the feedback. Please see [421532b](https://github.com/google/site-kit-wp/commit/421532b879570a1161be7ef0504b43ce90e0ff30) where I have added more supporting info via a SASS comment block, I trust this is what you meant.
> 
> Thanks!

Thanks @10upsimon, yes, that's spot on, cheers!

**Pull request merged**